lib/streamlit/elements/widgets/audio_input.py

```diff
@@ -0,0 +1,248 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from textwrap import dedent
+from typing import TYPE_CHECKING, Union, cast
+
+from typing_extensions import TypeAlias
+
+from streamlit.elements.form_utils import current_form_id
+from streamlit.elements.lib.policies import (
+    check_widget_policies,
+    maybe_raise_label_warnings,
+)
+from streamlit.elements.lib.utils import (
+    Key,
+    LabelVisibility,
+    compute_and_register_element_id,
+    get_label_visibility_proto_value,
+    to_key,
+)
+from streamlit.elements.widgets.file_uploader import _get_upload_files
+from streamlit.proto.AudioInput_pb2 import AudioInput as AudioInputProto
+from streamlit.proto.Common_pb2 import FileUploaderState as FileUploaderStateProto
+from streamlit.proto.Common_pb2 import UploadedFileInfo as UploadedFileInfoProto
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.runtime.scriptrunner import ScriptRunContext, get_script_run_ctx
+from streamlit.runtime.state import (
+    WidgetArgs,
+    WidgetCallback,
+    WidgetKwargs,
+    register_widget,
+)
+from streamlit.runtime.uploaded_file_manager import DeletedFile, UploadedFile
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+SomeUploadedAudioFile: TypeAlias = Union[UploadedFile, DeletedFile, None]
+
+
+@dataclass
+class AudioInputSerde:
+    def serialize(
+        self,
+        audio_file: SomeUploadedAudioFile,
+    ) -> FileUploaderStateProto:
+        state_proto = FileUploaderStateProto()
+
+        if audio_file is None or isinstance(audio_file, DeletedFile):
+            return state_proto
+
+        file_info: UploadedFileInfoProto = state_proto.uploaded_file_info.add()
+        file_info.file_id = audio_file.file_id
+        file_info.name = audio_file.name
+        file_info.size = audio_file.size
+        file_info.file_urls.CopyFrom(audio_file._file_urls)
+
+        return state_proto
+
+    def deserialize(
+        self, ui_value: FileUploaderStateProto | None, widget_id: str
+    ) -> SomeUploadedAudioFile:
+        upload_files = _get_upload_files(ui_value)
+        if len(upload_files) == 0:
+            return_value = None
+        else:
+            return_value = upload_files[0]
+        return return_value
+
+
+class AudioInputMixin:
+    @gather_metrics("audio_input")
+    def audio_input(
+        self,
+        label: str,
+        key: Key | None = None,
+        on_change: WidgetCallback | None = None,
+        args: WidgetArgs | None = None,
+        kwargs: WidgetKwargs | None = None,
+        *,  # keyword-only arguments:
```

Comment:
super nit: I think `# keyword-only arguments:` is not needed as this is not an exotic Python construct here so I personally would leave it out because we probably don't comment it anywhere else in the codebase